prql-dotnet/PrqlCompiler/PrqlCompiler.cs

```diff
@@ -38,53 +36,95 @@ public static string Compile(string prqlQuery)
         /// <exception cref="ArgumentException"><paramref name="prqlQuery"/> is null or empty.</exception>
         /// <exception cref="ArgumentNullException"><paramref name="options"/> is <c>null</c>.</exception>
         /// <exception cref="FormatException"><paramref name="prqlQuery"/> cannot be compiled.</exception>
-        public static string Compile(string prqlQuery, PrqlCompilerOptions options)
+        public static Result Compile(string prqlQuery, PrqlCompilerOptions options)
         {
             if (string.IsNullOrEmpty(prqlQuery))
             {
                 throw new ArgumentException(nameof(prqlQuery));
```

Comment:
Is this name supposed to match the name at https://github.com/PRQL/prql/pull/2111/files#diff-628f74a4e1fbc63c5979f552afd14a7043936043a54700da168b3186fa36dc22L39? One has `Null` in...

Reply:
It documents the guard clause which throws if the input parameter `options` is `null`. If any property inside the argument is null that is a different matter.

Reply:
Ah yes I see, `ArgumentException` is also on L36. Thanks!